File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.Arrays;

class Solution {
    public long solution(long n) {
        StringBuilder sb = new StringBuilder();
        String str = String.valueOf(n);
        int len = str.length();
        int[] arr = new int[len];
        
        for(int i = 0; i < len; i++) {
            arr[i] = str.charAt(i) - '0';
        }

        Arrays.sort(arr);
    
        for(int i = 0; i < len; i++) {
           sb.append(arr[len-i-1]);
        }

        return Long.parseLong(sb.toString());
    }
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.py
def solution(n):
    list = []
    while n > 0:
        list.append(n%10)
        n //= 10
        
    list.sort()
    list.reverse()
    
    s = ""
    for x in list:
        s += str(x)
        
    return int(s)